src/jit/valuenum.cpp

```diff
@@ -6316,7 +6316,10 @@ ValueNum Compiler::fgMemoryVNForLoopSideEffects(MemoryKind  memoryKind,
                 if (verbose)
                 {
                     var_types elemTyp = DecodeElemType(elemClsHnd);
-                    if (varTypeIsStruct(elemTyp))
+                    // If a valid class handle is given when the ElemType is set, DecodeElemType will
+                    // return TYP_STRUCT, and elemClsHnd is that handle.
+                    // Otherwise, elemClsHnd is NOT a valid class handle, and is the encoded var_types value.
+                    if (elemTyp == TYP_STRUCT)
```

Comment:
I thought I'd commented on this before, but it doesn't seem that I did. Although this is `DEBUG` only code, it impacts SIMD testing on desktop which unfortunately relies on the jitdump. In the case where we don't have a valid handle for a SIMD type, the `eeGetClassName` will get a VM assert.

Reply:
And FWIW, this behavior also exists in coreclr. Doing this:
```
set COMPlus_TieredCompilation=0
set COMPlus_JitDump=VectorArray
Tests\Core_Root\CoreRun.exe JIT\SIMD\VectorArray_ro\VectorArray_ro.exe > jitdump.out
```
Produces the following without this fix:
```
CORECLR! TypeDesc::IsRestored_NoLogging + 0x0 (0x00007ffa`f63678fc)
CORECLR! TypeHandle::Verify + 0x1D (0x00007ffa`f6316ab9)
CORECLR! CEEInfo::getClassName + 0x197 (0x00007ffa`f6195357)
CLRJIT! Compiler::eeGetClassName + 0x34 (0x00007ffb`05501a58)
CLRJIT! Compiler::fgMemoryVNForLoopSideEffects + 0x36A (0x00007ffb`05614516)
CLRJIT! Compiler::fgValueNumberBlock + 0x7E5 (0x00007ffb`05616715)
...
```